components/service/glean/src/test/java/mozilla/components/service/glean/storages/StringsStorageEngineTest.kt

```diff
@@ -59,6 +59,29 @@ class StringsStorageEngineTest {
         assertEquals("test", snapshot["telemetry.valid"])
     }
 
+    @Test
+    fun `string serializer should correctly serialize strings`() {
+        val metric = StringMetricType(
+                disabled = false,
+                category = "telemetry",
+                lifetime = Lifetime.User,
+                name = "string_metric",
+                sendInPings = listOf("store1")
+        )
+
+        // Record the string in the store, without providing optional arguments.
+        StringsStorageEngine.record(
+                metric,
+                value = "test_string_value"
+        )
+
+        // Get the snapshot from "store1" and clear it.
+        val snapshot = StringsStorageEngine.getSnapshotAsJSON(storeName = "store1", clearStore = true)
```

Comment:
Ok, this is correctly calling the serialization function but it's checking the stored value against the cached data. In order to make sure that serialization is working you should probably either:

- manually load the SharedPreferences file and check the content (tricky)
- use two instances of the engine implementation, instead of using the singletone. Use a scoped variable to store data using the first one, then use a second instance and take a snapshot. See [this example](https://github.com/mozilla-mobile/android-components/blob/96e94ca5d653e040e3f2cfb26ac6ffae8a470e07/components/service/glean/src/test/java/mozilla/components/service/glean/storages/GenericScalarStorageEngineTest.kt#L369). I think this is the easiest way to go.

Reply:
This makes sense, thanks for the example.  Making corrections now.